CaloReco/src/CaloHitMakerFast_module.cc

```diff
@@ -31,148 +32,149 @@ namespace mu2e {
+      fhicl::Atom<double>        digiSampling       { Name("digiSampling"),       Comment("Digitization time sampling") }; 
+      fhicl::Atom<double>        deltaTPulses       { Name("deltaTPulses"),       Comment("Maximum time difference between two signals") }; 
+      fhicl::Atom<double>        nPEperMeV          { Name("nPEperMeV"),          Comment("number of photo-electrons per MeV") }; 
+      fhicl::Atom<double>        noiseLevelMeV      { Name("noiseLevelMeV"),      Comment("Noise level in MeV") }; 
+      fhicl::Atom<double>        nSigmaNoise        { Name("nSigmaNoise"),        Comment("Maxnumber of sigma Noise to combine digi") }; 
+      fhicl::Sequence<int>       caphriCrystalID    { Name("caphriCrystalID"),    Comment("List of caphri crystal ID") }; 
+      fhicl::Atom<int>           diagLevel          { Name("diagLevel"),          Comment("Diag Level"),0 };
     };
 
-
-
-    //--------------------------------------------------------------------------------------------------------------
-    void CaloHitMakerFast::produce(art::Event& event)
+    explicit CaloHitMakerFast(const art::EDProducer::Table<Config>& config) :
+      EDProducer{config},
+      caloDigisToken_    {consumes<CaloDigiCollection>(config().caloDigiCollection())},
+      pbttoken_          {consumes<ProtonBunchTime>(config().pbttoken())},
+      digiSampling_      (config().digiSampling()),
+      deltaTPulses_      (config().deltaTPulses()),
+      nPEperMeV_         (config().nPEperMeV()),
+      noise2_            (config().noiseLevelMeV()*config().noiseLevelMeV()),
+      nSigmaNoise_       (config().nSigmaNoise()),
+      caphriCrystalID_   (config().caphriCrystalID()),
+      diagLevel_         (config().diagLevel())
     {
-        if (diagLevel_ > 0) std::cout<<"[FastRecoDigiFromDigi] begin"<<std::endl;
-
-        auto pbtH = event.getValidHandle(pbttoken_);
-        const ProtonBunchTime& pbt(*pbtH);
-        double pbtOffset = pbt.pbtime_;
-
-        const auto& caloDigisHandle = event.getValidHandle(caloDigisToken_);
-        const auto& caloDigis(*caloDigisHandle);
-
-        auto caloHitsColl   = std::make_unique<CaloHitCollection>();
-        auto caphriHitsColl = std::make_unique<CaloHitCollection>();
-        extractHits(caloDigis,*caloHitsColl,*caphriHitsColl,pbtOffset);
-
-        event.put(std::move(caloHitsColl),  "calo");
-        event.put(std::move(caphriHitsColl),"caphri");
-
-        if (diagLevel_ > 0) std::cout<<"[FastRecoDigiFromDigi] end"<<std::endl;
-        return;
+      produces<CaloHitCollection>("calo");
+      produces<CaloHitCollection>("caphri");
+      produces<IntensityInfoCalo>();
     }
 
+    void produce(art::Event& e) override;
+
+  private:
+    using pulseMapType = std::unordered_map<unsigned, std::vector<HitInfo>>;
+    
+    void extractHits(const CaloDigiCollection& caloDigis, CaloHitCollection& caloHitsColl, CaloHitCollection& caphriHitsColl, , IntensityInfoCalo&, double pbtOffset);
```

Comment:
According to the build log this is the error is the two consecutive commas here.  I can't tell if that's the only problem.

Reply:
fixed